core/src/main/java/com/linecorp/armeria/server/http/auth/BasicToken.java

```diff
@@ -0,0 +1,90 @@
+package com.linecorp.armeria.server.http.auth;
+
+import static java.util.Objects.requireNonNull;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import java.util.Base64.Encoder;
+import java.util.Objects;
+
+/**
+ * The bearer token of <a href="https://en.wikipedia.org/wiki/Basic_access_authentication">HTTP Basic access authentication</a>.
+ * */
+public final class BasicToken {
+
+    private static final Encoder BASE64_ENCODER = Base64.getEncoder();
+    private static final Decoder BASE64_DECODER = Base64.getDecoder();
+
+    /**
+     * Creates a new {@link BasicToken} from given {@code username} and {@code password}.
+     * */
+    public static BasicToken of(String username, String password) {
+        return new BasicToken(username, password);
+    }
+
+    /**
+     * Creates a new {@link BasicToken} from the given base64-encoded string.
+     * */
+    public static BasicToken decode(String base64) {
+        String credential = new String(BASE64_DECODER.decode(base64), StandardCharsets.UTF_8);
+        int sep = credential.indexOf(':');
+        if (sep == -1) {
+            throw new IllegalArgumentException("Invalid credential: " + credential);
+        }
+        String username = credential.substring(0, sep);
+        String password = credential.substring(sep + 1);
+        return of(username, password);
+    }
+
+    private final String username;
+    private final String password;
+
+    private BasicToken(String username, String password) {
+        this.username = requireNonNull(username, "username");
+        this.password = requireNonNull(password, "password");
+    }
+
+    public String username() {
+        return username;
+    }
+
+    public String password() {
+        return password;
+    }
+
+    /**
+     * Returns HTTP header value.
+     * */
+    public String encode() {
+        return BASE64_ENCODER.encodeToString((username + ':' + password).getBytes(StandardCharsets.US_ASCII));
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        BasicToken that = (BasicToken) o;
+        return Objects.equals(username, that.username) &&
+               Objects.equals(password, that.password);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(username, password);
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder("BasicToken(username:");
+        sb.append(username);
+        sb.append(", password:");
+        sb.append(password);
```

Comment:
Perhaps we should not include password in `toString()` result due to security reasons?